plugins/modules/cloud/misc/proxmox_disk.py

```diff
@@ -0,0 +1,718 @@
+        config_str = disk_config["volume"]
+        attributes = self.get_create_update_attributes()
+
+        for k, v in attributes.items():
+            config_str += ',%s=%s' % (k, v)
+
+        # Now compare old and new config to detect if changes are needed
+        for option in ['size', 'storage_name', 'volume', 'volume_name']:
+            attributes.update({option: disk_config[option]})
+        # Values in params are numbers, but strings are needed to compare with disk_config
+        attributes = dict((k, str(v)) for k, v in attributes.items())
+        if disk_config == attributes:
+            return False
+
+        update_disk = {self.module.params["disk"]: config_str}
+        self.proxmox_api.nodes(vm['node']).qemu(vmid).config.set(**update_disk)
+        return True
+
+    def move_disk(self, disk, vmid, vm, vm_config):
+        params = dict()
+        params['disk'] = disk
+        params['vmid'] = vmid
+        params['bwlimit'] = self.module.params['bwlimit']
+        params['storage'] = self.module.params['target_storage']
+        params['target-disk'] = self.module.params['target_disk']
+        params['target-vmid'] = self.module.params['target_vmid']
+        params['format'] = self.module.params['format']
+        params['delete'] = 1 if self.module.params.get('delete_moved', False) else 0
+        # Remove not defined args
+        params = dict((k, v) for k, v in params.items() if v is not None)
+
+        if params.get('storage', False):
+            disk_config = disk_conf_str_to_dict(vm_config[disk])
+            if params['storage'] == disk_config['storage_name']:
+                return False
+
+        taskid = self.proxmox_api.nodes(vm['node']).qemu(vmid).move_disk.post(**params)
+        timeout = self.module.params['timeout']
+        while timeout:
+            status_data = self.proxmox_api.nodes(vm['node']).tasks(taskid).status.get()
+            if status_data['status'] == 'stopped' and status_data['exitstatus'] == 'OK':
+                return True
+            if timeout <= 0:
+                self.module.fail_json(
+                    msg='Reached timeout while waiting for moving VM disk. Last line in task before timeout: %s' %
+                        self.proxmox_api.nodes(vm['node']).tasks(taskid).log.get()[:1])
+
+            sleep(1)
+            timeout -= 1
+        return True
+
+
+def main():
+    module_args = proxmox_auth_argument_spec()
+    disk_args = dict(
+        # Proxmox native parameters
+        aio=dict(choices=['native', 'threads', 'io_uring']),
+        backup=dict(type='bool'),
+        bps_max_length=dict(type='int'),
+        bps_rd_max_length=dict(type='int'),
+        bps_wr_max_length=dict(type='int'),
+        cache=dict(choices=['none', 'writethrough', 'writeback', 'unsafe', 'directsync']),
+        cyls=dict(type='int'),
+        detect_zeroes=dict(type='bool'),
+        discard=dict(choices=['ignore', 'on']),
```

Comment:
Also a good practice to be explicit about the type:
```suggestion
        discard=dict(type='str', choices=['ignore', 'on']),
```

Reply:
Thank you, added to all arguments